beacon_node/network/src/subnet_service/tests/mod.rs

```diff
@@ -338,110 +339,83 @@ mod attestation_service {
         assert_eq!(subnet_id1, subnet_id2);
 
         // submit the subscriptions
-        attestation_service
+        subnet_service
             .validator_subscriptions(vec![sub1, sub2].into_iter())
             .unwrap();
 
         // Unsubscription event should happen at slot 2 (since subnet id's are the same, unsubscription event should be at higher slot + 1)
-        // Get all events for 1 slot duration (unsubscription event should happen after 2 slot durations).
-        let events = get_events(&mut attestation_service, None, 1).await;
-        matches::assert_matches!(
-            events[..3],
-            [
-                SubnetServiceMessage::Subscribe(_any1),
-                SubnetServiceMessage::EnrAdd(_any3),
-                SubnetServiceMessage::DiscoverPeers(_),
-            ]
-        );
-
         let expected = SubnetServiceMessage::Subscribe(Subnet::Attestation(subnet_id1));
 
-        // Should be still subscribed to 2 long lived and up to 1 short lived subnet if both are
-        // different.
-        if !attestation_service.is_subscribed(
-            &subnet_id1,
-            attestation_subnets::SubscriptionKind::LongLived,
-        ) {
-            // The index is 3*subnets_per_node (because we subscribe + discover + enr per long lived
-            // subnet) + 1
-            let index = 3 * subnets_per_node;
-            assert_eq!(expected, events[index]);
-            assert_eq!(
-                attestation_service.subscription_count(),
-                subnets_per_node + 1
-            );
+        if subnet_service.is_subscribed(&Subnet::Attestation(subnet_id1)) {
+            // If we are permanently subscribed to this subnet, we won't see a subscribe message
+            let _ = get_events(&mut subnet_service, None, 1).await;
         } else {
-            assert!(attestation_service.subscription_count() == subnets_per_node);
+            let subscription = get_events(&mut subnet_service, None, 1).await;
+            assert_eq!(subscription, [expected]);
```

Comment:
This test is currently failing here
https://github.com/sigp/lighthouse/actions/runs/10933145564/job/30351144513?pr=6146

Reply:
This is now passing in the 2nd run and on my local env 🤔 
Is there a potential timing issue here?

Reply:
Yeah, there is a timeout on the `get_events`. Ideally I wanted to move this to a deterministic thing rather than times, but I couldn't find a good way to do it. 

This is why we're using timeouts in the get_events() and making them fairly large. I think this test could be improved to not rely on a single slot tho. 

I haven't done it in this PR. If the test keeps failing, we'll look into it I rekon

Reply:
Sounds good